app/models/user.rb

```diff
@@ -4,17 +4,40 @@ class User < ActiveRecord::Base
   validates :payment_url, url: { allow_blank: true, no_local: true, schemes: %w(http https) }
   validates :duns_number, duns_number: true
   validates :email, presence: true, email: true
-  validates :github_id, presence: true
-  validates :github_login, presence: true
+  validates :github_id, presence: true, if: :not_login_user?
+  validates :github_login, presence: true, if: :not_login_user?
   validates :sam_status, presence: true
 
   enum sam_status: { duns_blank: 0, sam_accepted: 1, sam_rejected: 2, sam_pending: 3 }
 
+  def self.from_omniauth(auth)
```

Comment:
note that once we actually let users sign in with login.gov, this will find an existing user via email rather than find or create. doing it this way for now so it's easier to test, but we can also make it just FIND if people are uncomfortable with this bandaid